py/htm/examples/mnist.py

```diff
@@ -132,17 +115,17 @@ def main(parameters=default_parameters, argv=None, verbose=True):
     # Training Loop
     for i in range(len(train_images)):
         img, lbl = random.choice(training_data)
-        enc.dense = img >= np.mean(img) # Convert greyscale image to binary.
+        encode(img, enc)
         sp.compute( enc, True, columns )
-        sdrc.learn( columns, lbl )
+        sdrc.learn( columns, lbl ) #TODO SDRClassifier could accept string as a label, currently must be int
```

Comment:
@ctrl-z-9000-times what do you think about passing `int(hash(lbl))` to c++ in python classifier? That way the label can be anything (hashable).

Reply:
That will not work.  The labels for the classifier need to be small integers since the classifier uses an array for them instead of a map.  If you give the SDRC the label "999" then the SDRC will make buckets for all the labels between 0 and 999.